host/frontend/host-orchestrator/operation.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"sync"
+)
+
+type NewOperationError string
+
+func (s NewOperationError) Error() string {
+	return fmt.Sprintf("failure creating a new operation: %s", string(s))
+}
+
+type OperationData struct {
```

Comment:
I suggested elsewhere to add a WaitGroup in this struct before looking at how it was used, some modifications are required to enable that:
- The operation manager needs to work with pointers to operations
- The OperationData structs needs to provide proper synchronization to access its fields since it's going to be shared among several threads
- Since the Operation only changes data from NotDone to Done, you can probably guarantee proper synchronization with just atomics and the correct order of updates (set result first and done later). Although a mutex would be correct too.

Reply:
acknowledged, moved the WaitGroup to the `MapOM` values, as I'd like to keep the Operation related structs to holding data only, I created an `operationEntry` type which is the one will have the WaitGroup and will managed with pointers within the `MapOM` scope only.